packages/autocomplete-js/src/types/AutocompleteSource.ts

```diff
@@ -5,15 +5,20 @@ import {
 } from '@algolia/autocomplete-core';
 
 import { AutocompleteComponents } from './AutocompleteComponents';
-import { AutocompleteRenderer, VNode } from './AutocompleteRenderer';
+import {
+  AutocompleteRenderer,
+  HTMLTemplate,
+  VNode,
+} from './AutocompleteRenderer';
 import { AutocompleteState } from './AutocompleteState';
 
 type Template<TParams> = (
   params: TParams &
     AutocompleteRenderer & {
       components: AutocompleteComponents;
+      html: HTMLTemplate;
     }
-) => VNode | string;
+) => VNode | VNode[] | string;
```

Comment:
`VNode[]` are valid children. This is necessary so any `html` template can be assignable to `Template`'s return type.